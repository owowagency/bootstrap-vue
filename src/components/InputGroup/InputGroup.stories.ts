import InputGroup from '.';
import {sizes} from '@/composables/useSize';

export default {
    title: 'Forms/Input group',
    component: InputGroup,
    argTypes: {
        append: {
            control: {type: 'text'},
        },
        prepend: {
            control: {type: 'text'},
        },
        size: {
            control: {type: 'select'},
            options: sizes,
        },
    },
};

const template = (args) => ({
    components: {InputGroup},
    setup: () => ({args}),
    template: '<InputGroup v-bind="args"><input class="form-control" type="number"></InputGroup>',
});

export const Default = template.bind({});

Default.args = {
    append: '.00',
    prepend: '€',
};

export const Sizing = (args) => ({
    components: {InputGroup},
    setup() {
        return {args, sizes};
    },
    template: '<InputGroup v-for="s of sizes" class="mb-1" :size="s" v-bind="args"><input class="form-control" type="number"></InputGroup>',
});

Sizing.args = {
    append: '.00',
    prepend: '€',
};
