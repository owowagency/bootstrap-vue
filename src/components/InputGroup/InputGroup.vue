<template>
    <div
        class="input-group"
        :class="classes"
    >
        <!-- @slot Displays the prepend of the input -->
        <slot name="prepend">
            <span
                v-if="prepend"
                class="input-group-text"
            >
                {{ prepend }}
            </span>
        </slot>


        <!-- @slot Displays the content of the input -->
        <slot />

        <!-- @slot Displays the append of the input -->
        <slot name="append">
            <span
                v-if="append"
                class="input-group-text"
            >
                {{ append }}
            </span>
        </slot>
    </div>
</template>

<script lang="ts">
import useSize, {sizeProps} from '@/composables/useSize';
import {computed} from 'vue';
import useClasses from '@/composables/useClasses';
</script>

<script lang="ts" setup>
const props = defineProps({
    /**
     * The text to prepend to the input
     */
    append: {
        type: String,
        default: undefined,
    },
    /**
     * The text to append to the input
     */
    prepend: {
        type: String,
        default: undefined,
    },
    ...sizeProps,
});

const {classes} = useClasses(computed(() => [
    useSize(props.size, 'input-group-{0}').sizeClass.value,
]));
</script>

<docs>
```vue
<InputGroup prepend="$">
    <FormControl v-model="price" />
</InputGroup>
```

## Imported Props

| Prop name | Description               | Type                                       | Values | Default |
| --------- | ------------------------- | ------------------------------------------ | ------ | ------- |
| size      | The size of the component | [Size](../../composables/useSize) (string) | -      | 'md'    |
</docs>
