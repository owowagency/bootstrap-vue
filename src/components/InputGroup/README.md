# InputGroup

```vue
<InputGroup prepend="$">
    <FormControl v-model="price" />
</InputGroup>
```

## Imported Props

| Prop name | Description               | Type                                       | Values | Default |
| --------- | ------------------------- | ------------------------------------------ | ------ | ------- |
| size      | The size of the component | [Size](../../composables/useSize) (string) | -      | 'md'    |

## Props

| Prop name | Description                      | Type   | Values | Default   |
| --------- | -------------------------------- | ------ | ------ | --------- |
| append    | The text to prepend to the input | string | -      | undefined |
| prepend   | The text to append to the input  | string | -      | undefined |

## Slots

| Name    | Description                       | Bindings |
| ------- | --------------------------------- | -------- |
| prepend | Displays the prepend of the input |          |
| default | Displays the content of the input |          |
| append  | Displays the append of the input  |          |
