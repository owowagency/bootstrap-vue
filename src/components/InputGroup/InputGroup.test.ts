import {describe, expect, it} from 'vitest';
import InputGroup from '.';
import {shallowMount} from '@vue/test-utils';

const textPropTest = (propName: string) => {
    it(`renders ${propName}`, () => {
        const text = `${propName} me`;

        const wrapper = shallowMount(InputGroup, {
            props: {[propName]: text},
        });

        expect(wrapper.find('.input-group-text').text()).toBe(text);
    });
};

describe('template', () => {
    componentRenderTest(InputGroup);

    componentWrapperClassTest(InputGroup, {size: 'lg'}, 'input-group-lg');

    componentSlotRenderTest(InputGroup, 'append');

    componentSlotRenderTest(InputGroup);

    componentSlotRenderTest(InputGroup, 'prepend');

    ['append', 'prepend'].forEach(textPropTest);
});
