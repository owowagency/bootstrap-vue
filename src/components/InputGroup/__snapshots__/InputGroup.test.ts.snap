// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`template > renders default 1`] = `
"<div class=\\"input-group input-group-md\\">
  <!-- @slot Displays the prepend of the input -->
  <!--v-if-->
  <!-- @slot Displays the content of the input -->
  <!-- @slot Displays the append of the input -->
  <!--v-if-->
</div>"
`;
